crates/apub/src/lib.rs

```diff
@@ -69,16 +69,21 @@ fn check_apub_id_valid(apub_id: &Url, local_site_data: &LocalSiteData) -> Result
     return Err("Federation disabled");
   }
 
-  if let Some(blocked) = local_site_data.blocked_instances.as_ref() {
-    if blocked.iter().any(|i| domain.eq(&i.domain)) {
-      return Err("Domain is blocked");
-    }
+  if local_site_data
+    .blocked_instances
+    .iter()
+    .any(|i| domain.eq(&i.domain))
+  {
+    return Err("Domain is blocked");
   }
 
-  if let Some(allowed) = local_site_data.allowed_instances.as_ref() {
-    if !allowed.iter().any(|i| domain.eq(&i.domain)) {
-      return Err("Domain is not in allowlist");
-    }
+  if !local_site_data.allowed_instances.is_empty()
+    && !local_site_data
+      .allowed_instances
+      .iter()
+      .any(|i| domain.eq(&i.domain))
```

Comment:
The is_empty check isnt necessary.

Reply:
I'd thought it wasn't but it fails otherwise. You can check the previous failed CI attempts for that.

Reply:
I think you need to change it to `if local_site_data.allowed_instances.iter().all(|i| domain.ne(&i.domain)) {`

Reply:
That would check to make sure **every instance** in that allowlist is the domain to be checked. We only want to see if the one to be checked is there or not.

Reply:
Ah the is_empty is actually to check if allowlist mode is enabled, so it does require a separate check. Could you add a comment to explain that?

Reply:
Done

Reply:
Didnt get pushed?